app/src/main/java/nu/yona/app/utils/AppUtils.java

```diff
@@ -626,37 +627,35 @@ private static Intent getVPNIntent(VpnProfile profile, Context context)
 	public static void downloadCertificates()
 	{
 		User user = YonaApplication.getEventChangeManager().getDataState().getUser();
-		if (user != null && user.getLinks() != null)
+		if (user == null || user.getEmbedded() == null || user.getEmbedded().getYonaDevices() == null
```

Comment:
What does this do? IMHO, the check should be: ``if (!user.isMobileNumberConfirmed())``.
The other checks hide inconsistencies.